src/main/java/com/github/onsdigital/perkin/publish/FtpPublisher.java

```diff
@@ -41,15 +45,17 @@ public void publish(List<DataFile> files) throws IOException {
 
             InputStream inputStream = new ByteArrayInputStream(file.getBytes());
 
-            String path = determinPath(file.getPath());
+            String path = determinePath(file.getPath());
             put(inputStream, path, filename);
 
             timer.stopStatus(200);
             Audit.getInstance().increment(timer, file);
         }
+
+        logout();
```

Comment:
i think logout() should be within a finally block - if put throws an IOException then the ftp connection will not be closed.
should add a test to show that when put throws an IOException, ftp connection is closed